sdk/devcenter/Azure.ResourceManager.DevCenter/src/Generated/Models/DevCenterOperationStatus.Serialization.cs

```diff
@@ -58,18 +58,39 @@ void IJsonModel<DevCenterOperationStatus>.Write(Utf8JsonWriter writer, ModelRead
             writer.WriteStringValue(Status);
             if (Optional.IsDefined(PercentComplete))
             {
-                writer.WritePropertyName("percentComplete"u8);
-                writer.WriteNumberValue(PercentComplete.Value);
+                if (PercentComplete != null)
+                {
+                    writer.WritePropertyName("percentComplete"u8);
+                    writer.WriteNumberValue(PercentComplete.Value);
+                }
+                else
+                {
+                    writer.WriteNull("percentComplete");
+                }
```

Comment:
this actually changes nothing.
The type of `PercentComplete` is `int?`. this part is wrapped inside a `Optional.IsDefined` condition.
This is how `Optional.IsDefined` is implemented:
```csharp
public static bool IsDefined<T>(T? value) where T : struct
{
	return value.HasValue;
}
```
Therefore this piece of code is equivalent to this:
```csharp
if (PercentComplete.HasValue)
{
	if (PercentComplete != null)
	{
		// do old things
	}
	else
	{
		// do new things
	}
}
```
For nullable structs, `!= null` is the grammar sugar of `HasValue` and they are identical. Therefore this piece of code is checking the same condition twice therefore as a result, the old things will still be executed, and there is no situation that we could go and execute the new things.

This is how "this code change actually changes nothing".

Reply:
The reason why we have this change, is that this model is special, that it inherits a model from resource manager, and this property `PercentComplete` also comes from that base model.
[This generator change](https://github.com/Azure/autorest.csharp/pull/4816) is changing how we parse the nullable struct types such as `int?`. Previously it is incorrectly parsed as `Nullable<int>` and the `IsNullable` flag on this result is actually false, therefore previously the code is generated as if it is not nullable.
Now the generator change fixes it, the code is generated properly as the property is nullable.

But in fact, because of the shortage of dotnet or our design, we represent optionality from nullability therefore we could never write a "optional and nullable" property entirely correctly.

Reply:
Therefore based on the two reasons above, I would say this change is by-design and follows all the rules we already have in the generator.